media/js/mozorg/diversity/2021/diversity-2021.es6.js

```diff
@@ -0,0 +1,198 @@
+    </div>`;
+
+// Set to true when a listener it set, as to not set it multiple times.
+// Resets when the modal is closed.
+let kbInit = null;
+
+function keyboardListenerInit() {
+    if (kbInit) {
+        return;
+        // Turn Off
+    }
+
+    document.addEventListener('keyup', keyboardNextPrev, false);
+    kbInit = true;
+}
+
+function keyboardNextPrev(event) {
+    if (!kbInit) {
+        return;
+    }
+
+    switch (event.keyCode) {
+        case 37: // Left arrow
+            prevModalArticle();
+            break;
+        case 38: // Up arrow
+            prevModalArticle();
+            break;
+        case 39: // Right arrow
+            nextModalArticle();
+            break;
+        case 40: // Down arrow
+            nextModalArticle();
+            break;
+    }
+}
+
+function modalInit() {
+    const modalNextButton = document.querySelector('.c-modal-next');
+    const modalPrevButton = document.querySelector('.c-modal-prev');
+
+    modalNextButton.removeEventListener('click', nextModalArticle, false);
+    modalPrevButton.removeEventListener('click', prevModalArticle, false);
+
+    modalNextButton.addEventListener('click', nextModalArticle, false);
+    modalPrevButton.addEventListener('click', prevModalArticle, false);
+
+    keyboardListenerInit();
+}
+
+function getCurrentModalIndex() {
+    const modalContent = document.querySelector(
+        '.mzp-u-modal-content.mzp-c-modal-overlay-contents'
+    );
+    const newArticleIndex = parseInt(
+        modalContent.getAttribute('data-current-index'),
+        10
+    );
+    return newArticleIndex;
+}
+
+function updateModalArticle(index) {
+    const modalContent = document.querySelector(
+        '.mzp-u-modal-content.mzp-c-modal-overlay-contents'
+    );
+    const newArticleId = articleArray[index].getAttribute('data-modal-id');
+    const newModalContent = document
+        .querySelector(`[data-modal-parent="${newArticleId}"]`)
+        .cloneNode(true);
+    const currentModalContent = modalContent.firstElementChild;
+
+    window.location.hash = newArticleId;
+
+    modalContent.replaceChild(newModalContent, currentModalContent);
+    modalContent.setAttribute('data-current-index', index);
+
+    modalInit();
+}
+
+function nextModalArticle() {
+    let newArticleIndex = getCurrentModalIndex();
+    newArticleIndex++;
+
+    // If at the end of the gallery, start over
+    if (newArticleIndex === articleArray.length) {
+        newArticleIndex = 0;
+    }
+
+    updateModalArticle(newArticleIndex);
+}
+
+function prevModalArticle() {
+    let newArticleIndex = getCurrentModalIndex();
+    newArticleIndex--;
+
+    // If at the beginning of the gallery, start over
+    if (newArticleIndex < 0) {
+        newArticleIndex = articleArray.length - 1;
+    }
+
+    updateModalArticle(newArticleIndex);
+}
+
+for (let i = 0; i < modalContainers.length; i++) {
+    const modalContainer = modalContainers[i];
+
+    modalContainer.setAttribute('aria-role', 'button');
```

Comment:
**suggestion (notsureif blocker):** was getting an error through axeDevTools for these sections: `ARIA attributes must conform to valid names` - not sure, but i _think_ that instead of aria-role you should be using `role` here.

https://rocketvalidator.com/accessibility-validation/aria-valid-attr

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role